core/embed/rust/src/ui/display/mod.rs

```diff
@@ -41,25 +41,27 @@ pub fn fade_backlight(target: u8) {
 
 #[cfg(feature = "backlight")]
 pub fn fade_backlight_duration(target: u8, duration_ms: u32) {
-    let target = target as i32;
-    let current = backlight() as i32;
+    let current = backlight();
     let duration = Duration::from_millis(duration_ms);
 
     if animation_disabled() {
-        set_backlight(target as u8);
+        set_backlight(target);
         return;
     }
 
     let timer = Stopwatch::new_started();
 
-    while timer.elapsed() < duration {
+    loop {
         let elapsed = timer.elapsed();
-        let val = i32::lerp(current, target, elapsed / duration);
-        set_backlight(val as u8);
+        if elapsed >= duration {
+            break;
+        }
+        let val = u8::lerp(current, target, elapsed / duration);
```

Comment:
please test the fading on the device - something tells me that previous approach might have been more smooth, but i might as well be wrong

Reply:
I think the fading looks similar (comparing 2.8.7 release and the firmware from this PR on T3T1):

https://github.com/user-attachments/assets/dd8e1202-77ef-40f7-879e-1a940fe4cf26


https://github.com/user-attachments/assets/45e4e1aa-353e-4047-ab67-f7929a3c0fb2